core/server/master/src/test/java/alluxio/master/AlluxioMasterProcessTest.java

```diff
@@ -209,17 +250,33 @@ public void restoreFromBackupLocal() throws Exception {
   }
 
   private void startStopTest(AlluxioMasterProcess master) throws Exception {
+    startStopTest(master, true, true);
+  }
+
+  private void startStopTest(AlluxioMasterProcess master,
+      boolean expectWebServiceStarted, boolean expectMetricsSinkStarted) throws Exception {
+    waitForAllServingReady(master, 5000);
+    assertTrue(expectWebServiceStarted == (master.getWebAddress() != null));
+    assertTrue(expectMetricsSinkStarted == MetricsSystem.isStarted());
```

Comment:
Why is `MetricsSystem.isStarted()` not using the `master` reference at all? I think you should add some methods to the master interface just like `MasterProcess.isServing()`/

So `MasterProcess.isServing()` -> `MasterProcess.isGrpcServing()` and you add `MasterProcess.isWebServing()` etc.

Reply:
Done.